src/Components/Profile/PreferenceBlock.tsx

```diff
@@ -0,0 +1,150 @@
+    const checked = Array.from(
+      document.querySelectorAll(`input[type=checkbox][name="course-checkbox-${course_instance.section_id}"]:checked`)
+    )
+    
+    checked
+      .map((e: Element) => parseInt(e.getAttribute('data-sid')!))
+      .forEach(id => {
+        userPrefs.set(id, value);
+      })
+
+    checked.forEach((e: any) => e.checked = false);
+    
+    setUserPrefs(new Map(userPrefs));
+  }
+
+  const resStatusColor = (section_id: number) => {
+    const status = userPrefs.get(section_id);
+    if (status === undefined) return statusColors.get("Indifferent")!;
+    return statusColors.get(status)!;
+  }
+  
+  const resStatusText = (section_id: number) => {
+    const status = userPrefs.get(section_id);
+    switch (status) {
+      case "Can't Do": return "(Can't)";
+      case "Prefer Not To Do": return "(Don't Want)";
+      case "Indifferent": return "";
+      case "Prefer To Do": return "(Want)";
+    }
+  }
+
+  const checkAll = (id: number) => {
+    Array.from(document.querySelectorAll(`input[type=checkbox][name="course-checkbox-${id}"]`)).forEach((e: any) => e.checked = true);
+  }
+
+  let formElements: JSX.Element[] = [];
+  let dotElements: JSX.Element[] = [];
+  let seenDots: APIUserPreferenceEnum[] = [];
+  course_instance.section_ids.forEach((section_id, i) => {
+    formElements.push(
+      <div key={`pref-row-${JSON.stringify(course_instance)}-${section_id}`} className="pref-row">
+        <input id={`pref-row-checkbox-${course_instance.course_number}-${section_id}`} type="checkbox" name={`course-checkbox-${course_instance.section_id}`} data-sid={section_id} />
+        <label htmlFor={`pref-row-checkbox-${course_instance.course_number}-${section_id}`} style={{color: resStatusColor(section_id)}} >
+          {course_instance.course_number}-{course_instance.section_numbers[i]} {resStatusText(section_id)}
+        </label>
+      </div>
+    )
+
+    const pref = userPrefs.get(section_id);
+    if (pref === "Indifferent" || pref === undefined || seenDots.includes(pref)) return;
+    
+    dotElements.push(
+      <div key={`pref-dot-${JSON.stringify(course_instance)}-${section_id}`} className="pref-dot" style={{backgroundColor: resStatusColor(section_id)}}/>
+    )
+    seenDots.push(userPrefs.get(section_id)!);
+  })
+  
+
+  let flex = calcFlex(visible, inline, size);
+  const isVisible = {
+    width: (visible)? undefined : 0,
+    flex: flex,
+    margin: visible? undefined : 0
+  }
+  
+  let color = { background: blockColors.get(course_instance.course_number)! };
+  let classes = "block";
+  if (linkIDs !== null && linkIDs.length === 0) classes += " alert";
+  if (interacted) {
+    classes += " interacted " + edge;
+    classes += (bottom)? ' bottom' : '';
+  }
+  else classes += " grow-h";
+
+  return (
+    <div ref={ref} className={classes} onClick={() => setInteracted(true)}
+      title={`${course_instance.course_number}-${course_instance.section_number}`} 
```

Comment:
Can we also display the professor's name here? It's stored in the DB as `placeholder_professor_name` and should likely be returned by an API somewhere.

Reply:
The idea for this, by the way, is that the professor teaching a course is often a motivator. We added that column to the DB for this purpose in particular.